vision/cifar10/cifar10.jl

```diff
@@ -158,21 +145,49 @@ function train(; kws...)
     loss(x, y) = logitcrossentropy(m(x), y)
 
     ## Training
-    # Defining the callback and the optimizer
-    evalcb = throttle(() -> @show(loss(val...)), args.throttle)
+    # Defining the optimizer
     opt = ADAM(args.lr)
+    ps = Flux.params(m)
+
     @info("Training....")
     # Starting to train models
-    Flux.@epochs args.epochs Flux.train!(loss, params(m), train, opt, cb = evalcb)
+    for epoch in 1:args.epochs
+        for (x, y) in train_data
+            x, y = gpu(x), gpu(y)
+
+            gs = Flux.gradient(ps) do 
+                loss(x, y)
+            end
+
+            Flux.update!(opt, ps, gs)
+        end
+
+        validation_loss = 0f0
+        for (x, y) in val_data
```

Comment:
Validation should also be batched I think

Reply:
What do you mean? val_data is a batched data loader, same as train_data, see line 139.